main.js

```diff
@@ -149,8 +152,11 @@ define(function (require, exports, module) {
     }
 
     function handleAutohideChange() {
-        if (prefs.get("autohide")) {
+        var document = DocumentManager.getCurrentDocument();
+        if (prefs.get("autohide") && isOutlineAvailable(document)) {
             Autohide.enable();
+        } else if (prefs.get("autohide") && !isOutlineAvailable(document)) {
+            OutlineManager.hideOutline();
```

Comment:
Is this branch really necessary? Wouldn't the outline be hidden when it is not available anyway?

Reply:
Original issue:

With Autohide activated, if we have an unsupported document open and we focus it, we can see the Autohide placeholder and if we hover, the empty Outline is displayed. Or the last one belonging to a supported document previously viewed.

This is solved in `handleEnabledChange ()` with
```
if (prefs.get ("autohide")) {
     Autohide.disable ();
}
```
but then a collateral issue appeared:

With Autohide activated, if we have a supported document open and an unsupported one too, we focus the supported one first. We deactivate Autohide and focus the unsupported. We can see that the Outline of the last supported file is shown (without Autohide).

That branch of the condition prevents precisely that.

Before this little hack, believe me I've been trying to correct these issues within the Autohide code itself. But it is too expensive. Maybe even to rethink the design and rewrite it.

Reply:
I do not know how I tested this fix the other day, but it's clear that I'm not very accurate. :disappointed:

I think that now everything must work correctly. Sorry.

Reply:
Thanks for the explanation, that makes sense.

I would be open to rethinking the design. Feel free to think about it and make a proposal if you like. 😃